### DISCUSSION TITLE
Solution on a mesh possible using Heat Conduction but not NS

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#17676] [Q&A General] [answered]
I have developed a large scale mesh problem that I was able to solve a conduction heat transfer problem on. When I apply the same mesh to a Naiver Stokes problem I am not able to solve the problem on the mesh domain. I am receiving the following error: Current number of dofs per element 72 is greater than AD_MAX_DOFS_PER_ELEM of 50. You can run `configure --with-derivative-size=<n
>` to request a larger derivative container.

The numerics associated with the problem are as follows:
Parallelism:
  Num Processors:          16
  Num Threads:             1
Mesh: 
  Parallel Type:           replicated
  Mesh Dimension:          3
  Spatial Dimension:       3
  Nodes:                   
    Total:                 5293752
    Local:                 325135
  Elems:                   
    Total:                 3222656
    Local:                 200035
  Num Subdomains:          5
  Num Partitions:          16
  Partitioner:             metis
Nonlinear System:
  AD size required:        72
  Num DOFs:                37056264
  Num Local DOFs:          2275945
  Variables:               "velocity" "p" "u" 
  Finite Element Types:    "LAGRANGE_VEC" "LAGRANGE" "LAGRANGE_VEC" 
  Approximation Orders:    "FIRST" "FIRST" "FIRST" 
Execution Information:
  Executioner:             Steady
  Solver Mode:             NEWTON
  MOOSE Preconditioner:    SMP (auto)

**Comment by cticenhour** [comment]
As the message suggested, you can run 

```
./configure --with-derivative-size=72
```

in your MOOSE directory, rebuild MOOSE, and then try to run your problem again. This command reconfigures MOOSE to use a larger container for the AD derivatives used in your calculation.

**Reply by TLWise** [reply]
Does this command work within the terminal command prompt of the moose directory, or is there a file that I will have to modify with this command?

Thank you for your continued attention to my issue.

**Reply by cticenhour** [reply]
You would run this command within the Terminal in your moose directory. When it completes, you'll then build and run moose (or your application using that copy of moose) as you usually do. Then you can try your problem again.

**Reply by TLWise** [reply]
Here are the two way I tried to implement what you described, along with the messages that I get.:

(base) [wisetamm@falcon1 moose]$ ./configure --with-derivative-size=72
-bash: ./configure: No such file or directory
 
(moose) [wisetamm@falcon1 moose]$  ./configure --with-derivative-size=72
-bash: ./configure: No such file or directory
 
Is it possible that I may need to do a moose update?

**Reply by cticenhour** [reply]
This is possible, as your copy of MOOSE would be quite old (older than around September 2020) if the `configure` file didn't exist in the MOOSE root directory. If you run the `git log` command in the MOOSE directory, you can see the date of your current version.

**Reply by cticenhour** [reply]
One more thing - if you wanted to keep your current version, the `configure` file should be in the `moose/framework` directory prior to September 2020. So you could try the configure command I mentioned above there as well.

**Reply by TLWise** [reply]
After running the configure command I am still receiving the max dof error. Here is the output after running the configure routine:

(moose) [wisetamm@falcon1 moose]$ ./configure --with-derivative-size=72
---------------------------------------------
----------- Configuring MOOSE ---------------
---------------------------------------------
configuring with derivative backing array size of 72
configuring with AD indexing type global
configuring with derivative type sparse
checking for pkg-config... /home/wisetamm/miniconda3/envs/moose/bin/pkg-config
configure: checking system for libpng
configuring with libpng support
configure: creating ./config.status
config.status: creating conf_vars.mk
config.status: creating framework/include/base/MooseConfig.h.tmp
config.status: framework/include/base/MooseConfig.h.tmp is unchanged
config.status: executing framework/include/base/MooseConfig.h commands
config.status: creating framework/include/base/MooseConfig.h - prefix MOOSE for framework/include/base/MooseConfig.h.tmp defines
config.status: framework/include/base/MooseConfig.h is unchanged
---------------------------------------------
--------- Done Configuring MOOSE ------------
---------------------------------------------

**Reply by GiudGiud** [reply]
72 might not be enough. Try 100 for the derivative size.
How many variables do you have in the system?

**Reply by TLWise** [reply]
given velocity depends on u,v, and w, it would appear that there are 5 variables being solved for.

**Reply by GiudGiud** [reply]
and they're all first order lagrange with what kind of element?
Please let us know if increasing the derivative size worked btw

**Reply by TLWise** [reply]
The nonlinear system information indicates the element type is LAGRANGE_VEC, LAGRANGE, LAGRANGE_VEC. I'll keep updates to my progress for this issue updated on this thread. Thank you for the advise and help.

**Reply by GiudGiud** [reply]
Oh I meant is this a tetrahedral or hexahedral mesh ?

**Reply by TLWise** [reply]
They are hex elements.

**Reply by TLWise** [reply]
Ran this configure but still got the error:
#(moose) [wisetamm@falcon1 moose]$ ./configure --with-derivative-size=100
#---------------------------------------------
#----------- Configuring MOOSE ---------------
#---------------------------------------------
#configuring with derivative backing array size of 100
#configuring with AD indexing type global
#configuring with derivative type sparse
#checking for pkg-config... /home/wisetamm/miniconda3/envs/moose/bin/pkg-config
#configure: checking system for libpng
#configuring with libpng support
#configure: creating ./config.status
#config.status: creating conf_vars.mk
#config.status: creating framework/include/base/MooseConfig.h.tmp
#config.status: executing framework/include/base/MooseConfig.h commands
#config.status: creating framework/include/base/MooseConfig.h - prefix MOOSE for framework/include/
#base/MooseConfig.h.tmp defines
#---------------------------------------------
#--------- Done Configuring MOOSE ------------
#---------------------------------------------

**Reply by TLWise** [reply]
Here is a copy of my .i file.
[FlowAboutRods.txt](https://github.com/idaholab/moose/files/6418549/FlowAboutRods.txt)

**Reply by TLWise** [reply]
The problem appears to be related to my hpc installation of moose. I did the configure process to change the number of DOFs on my desktop and the model ran with no errors concerning max number of DOFs. I will attempt to reinstall/update moose on the hpc to see if that eliminates the problem.

**Comment by TLWise** [comment, marked as answer]
Performed the moose update and and ran, ."/configure --with-derivative-size=100" in the moose directory. Did a re-make and test of the application. The error is no longer displaying.
